### PR TITLE
[3.10] Improve pre update checker cs and add a dedicated message when no extensions are found

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

[3.10] Improve pre update checker cs and add a dedicated message when no extensions are found

### Testing Instructions

- Install a plain 3.10-dev without extensions
- add this update server: https://update.joomla.org/core/nightlies/next_major_list.xml
- notice the message that no extensions have been found.

### Expected result

A message that no extensions have been found.

### Actual result

Only the notices part of the table with no indication what is happening or what this is reffering to.

### Documentation Changes Required

none

### CODE REVIEW

File: administrator/components/com_joomlaupdate/views/default/tmpl/default_preupdatecheck.php [hunk 1]
```diff
@@ -84,71 +80,79 @@
...
+					<li><?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_DESCRIPTION_BREAK'); ?></li>
+					<li><?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_DESCRIPTION_MISSING_TAG'); ?></li>
+					<li><?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_DESCRIPTION_UPDATE_REQUIRED'); ?></li>
+				</ul>
+			</fieldset>
+		</div>
+	</div>
+<?php else: ?>
+	<div class="row-fluid">
+		<fieldset class="span6">
```
Comment (brianteeman) [inline administrator/components/com_joomlaupdate/views/default/tmpl/default_preupdatecheck.php]: this markup is not correct as it is not a fieldset etc. Surely the markup should be the same as similar "no results" messages eg
```

		<?php if (empty($this->items)) : ?>
			<div class="alert alert-no-items">
				<?php echo JText::_('JGLOBAL_NO_MATCHING_RESULTS'); ?>
			</div>
		<?php else : ?>
```

File: administrator/language/en-GB/en-GB.com_joomlaupdate.ini [hunk 2]
```diff
@@ -55,6 +55,7 @@ COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSION_NAME="Extension Name"
 COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSION_VERSION_MISSING="Missing Compatibility Tag"
 COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSION_WARNING_UNKNOWN="Unkown error"
 COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS="Extensions Pre-Update Check"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_NONE="We have not detected any none-core extensions on this system."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_joomlaupdate.ini]: keep it simple

"No extensions installed."

Comment (brianteeman) [reply administrator/language/en-GB/en-GB.com_joomlaupdate.ini]: and don't let your colleagues see you use the word none-core or they will scream at you as I was screamed at for writing that

File: administrator/components/com_joomlaupdate/views/default/tmpl/default_preupdatecheck.php [hunk 3]
```diff
@@ -84,71 +80,79 @@
...
+					<li><?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_DESCRIPTION_MISSING_TAG'); ?></li>
+					<li><?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_DESCRIPTION_UPDATE_REQUIRED'); ?></li>
+				</ul>
+			</fieldset>
+		</div>
+	</div>
+<?php else: ?>
+	<div class="row-fluid">
+		<div class="span6">
+			<legend>
```
Comment (brianteeman) [inline administrator/components/com_joomlaupdate/views/default/tmpl/default_preupdatecheck.php]: without the fieldset the legend is invalid ;) 
I am not a pc with 3.10 installed right now so can only guess the code you need to change legend to some other title and then the p to 

```
<div class="alert alert-no-items">
<?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_NONE'); ?>
</div>
```

**Comment by brianteeman** [comment]
It works but the text looks really "lost"

image

It looks better (I believe) like this

image

The code for that is
```
		<div class="span6">
			<h3>
				<?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS'); ?>
			</h3>
			<div class="alert alert-no-items">
				<?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_NONE'); ?>
			</div>
		</div>
```

**Comment by zero-24** [comment]
Pushed thanks @brianteeman

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on e4d64edafcb9de957859005d76a5b0069b69ab4b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27316">issues.joomla.org/tracker/joomla-cms/27316</a>.</sub>

File: administrator/components/com_joomlaupdate/views/default/tmpl/default_preupdatecheck.php [hunk 4]
```diff
@@ -84,71 +80,79 @@
...
+					<li><?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_DESCRIPTION_BREAK'); ?></li>
+					<li><?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_DESCRIPTION_MISSING_TAG'); ?></li>
+					<li><?php echo JText::_('COM_JOOMLAUPDATE_VIEW_DEFAULT_DESCRIPTION_UPDATE_REQUIRED'); ?></li>
+				</ul>
+			</fieldset>
+		</div>
+	</div>
+<?php else: ?>
+	<div class="row-fluid">
+	<div class="span6">
```
Comment (Quy) [inline administrator/components/com_joomlaupdate/views/default/tmpl/default_preupdatecheck.php]:
```suggestion
		<div class="span6">
```

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on c3bf4b31f49a6252ebbea1ff5d7f48c84619a2de<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27316">issues.joomla.org/tracker/joomla-cms/27316</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27316">issues.joomla.org/tracker/joomla-cms/27316</a>.</sub>

**Comment by richard67** [comment]
Brian's test was still valid because after that was only one small code style change => RTC.

**Comment by zero-24** [comment]
Thanks :+1:

**Pull request merged**